Signal/src/ViewControllers/SignalsViewController.m

```diff
@@ -530,7 +532,11 @@ - (void)displayAnyUnseenUpgradeExperience
                          completion:^{
                              [self markAllUpgradeExperiencesAsSeen];
                          }];
+    } else if (!self.hasBeenPresented && [ProfileViewController shouldDisplayProfileViewOnLaunch]) {
+        [ProfileViewController presentForUpgradeOrNag:self];
```

Comment:
So, this does the bare minimum to arbitrate between "on launch" conflicts. Basically we only show the profile nag if there is no "upgrade experience" to be seen.